### PR TITLE
update k8s libs to 1.25.z and ginkgo to v2

### PR DESCRIPTION
**Pull request by fromanirh**
This PR bundles update to k8s libs 1.25.z and to ginkgo v2
We do these two big changes in one go because:

- kube 1.25 pulls ginkgo v2
- we cannot have ginkgo v1 and v2 in the same binary

so splitting the changes is not really easier than doing everything in one go.

### CODE REVIEW

**Comment by swatisehgal** [comment]
@fromanirh Is this still WIP?

**Comment by fromanirh** [comment]
> @fromanirh Is this still WIP?

just fixed :) now is reviewable.
We need to sync with @shajmakh and @mrniranjan to merge this PR with minimal disruption for the consumer of the tests though.

**Comment by fromanirh** [comment]
/hold
because https://github.com/openshift-kni/numaresources-operator/pull/466#issuecomment-1305489954

**Review by swatisehgal — APPROVED** [review]
/lgtm

Will let @shajmakh or @mrniranjan unhold the PR.

File: test/e2e/serial/e2e_serial_test.go [hunk 1]
```diff
@@ -22,13 +22,22 @@ import (
...
+	qe_reporters "kubevirt.io/qe-tools/pkg/ginkgo-reporters"
+
 	serialconfig "github.com/openshift-kni/numaresources-operator/test/e2e/serial/config"
 	_ "github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests"
 )
 
+var afterSuiteReporters = []Reporter{}
+
 func TestSerial(t *testing.T) {
+	if qe_reporters.Polarion.Run {
```
Comment (shajmakh) [inline test/e2e/serial/e2e_serial_test.go:37]: where should this argument be set? I see it is false by default

**Review by shajmakh — COMMENTED** [review]
thanks for all the changes!
left two comments for clarification.

File: hack/run-test-e2e.sh [hunk 2]
```diff
@@ -19,16 +19,16 @@ function test_sched() {
...
   # -requireSuite: fail if tests are not executed because of missing suite
-  ${BIN_DIR}/e2e-nrop-sched.test ${NO_COLOR} --ginkgo.v --ginkgo.failFast --ginkgo.flakeAttempts=2 --ginkgo.reportFile=/tmp/artifacts/nrop/e2e-sched
+  ${BIN_DIR}/e2e-nrop-sched.test ${NO_COLOR} --ginkgo.v --ginkgo.fail-fast --ginkgo.flake-attempts=2 --ginkgo.reportFile=/tmp/artifacts/nrop/e2e-sched
 }
 
 NO_COLOR=""
 if ! which tput &> /dev/null 2>&1 || [[ $(tput -T$TERM colors) -lt 8 ]]; then
   echo "Terminal does not seem to support colored output, disabling it"
-  NO_COLOR="-ginkgo.noColor"
+  NO_COLOR="-ginkgo.no-color"
```
[the remark this answers is not in the document]
Comment (fromanirh) [reply hack/run-test-e2e.sh:31]: all flags have been renamed: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#changed-command-line-flags

File: test/e2e/serial/e2e_serial_test.go (hunk 1 above)
Comment (fromanirh) [reply test/e2e/serial/e2e_serial_test.go:37]: The module is adding the flags at init time. I prefer a more explicit approach, but still this is a common pattern in the golang world:
https://github.com/kubevirt/qe-tools/blob/main/pkg/ginkgo-reporters/README.md
https://github.com/kubevirt/qe-tools/blob/main/pkg/ginkgo-reporters/polarion_reporter.go#L37

**Pull request merged**